### PR TITLE
Issue#2291: Added javadoc to the newly added public APIs

### PR DESCRIPTION
**Pull request by saurabhagas**
Added javadoc.

### CODE REVIEW

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java [hunk 1]
```diff
@@ -60,6 +60,14 @@ public ClusteringServiceConfiguration(URI clusterUri) {
     this(clusterUri, Timeouts.DEFAULT);
   }
 
+  /**
+   * Creates a {@code ClusteringServiceConfiguration} from the properties provided.
+   *
+   * @param servers the non-{@code null} Iterable<InetSocketAddress> identifying the servers in the cluster
```
Comment (chrisdennis) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: I think stating the type like that is a little verbose maybe make it a little more prose like: `an iterable of the servers in the cluster`?

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java [hunk 2]
```diff
@@ -60,6 +60,14 @@ public ClusteringServiceConfiguration(URI clusterUri) {
...
+   * @param clusterTierManager the non-{@code null} clusterTierManager identifying the cache manager
```
Comment (chrisdennis) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: This (and the other instances) is a little too close to referring to the cache manager name.

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java (hunk 1 above)
Comment (saurabhagas) [reply clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: Done

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java (hunk 2 above)
Comment (saurabhagas) [reply clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: Changed.

**Pull request merged**